signer/src/transaction_signer.rs

```diff
@@ -684,18 +786,144 @@ where
+                            "processing message"
+                        );
+
+                        let accepted_sighash =
+                            Self::validate_bitcoin_sign_request(&db, &request.message).await?;
+
+                        accepted_sighash.sighash.into()
+                    }
+                    WstsMessageId::DkgVerification(key) => {
+                        // This is a DKG verification signing round. The data
+                        // provided by the coordinator for signing is expected
+                        // to be the current bitcoin chain tip block hash, which
+                        // we validate and return an error if it does not match
+                        // our view of the current chain tip. We also verify
+                        // that the provided aggregate key matches our latest
+                        // aggregate key.
+
+                        let new_key: PublicKeyXOnly = key.into();
+
+                        // Validate the received message.
+                        Self::validate_dkg_verification_message(
+                            &db,
+                            &new_key,
+                            chain_tip,
+                            &request.message,
+                        )
+                        .await?;
+
+                        tracing::debug!(
+                            signature_type = ?request.signature_type,
+                            "processing message"
+                        );
+
+                        let key = key.into();
+                        let state_machine_id = self
+                            .ensure_dkg_verification_state_machine(chain_tip.block_hash, key)
+                            .await?;
+                        self.process_dkg_verification_message(key, state_machine_id, &msg.inner)
+                            .await?;
+                        state_machine_id
+                    }
+                };
+
                 let response = self
                     .relay_message(id, msg.id, &msg.inner, &chain_tip.block_hash)
                     .await;
 
                 self.wsts_state_machines.pop(&id);
                 response?;
             }
-            WstsNetMessage::NonceResponse(_) | WstsNetMessage::SignatureShareResponse(_) => {}
+            WstsNetMessage::NonceResponse(request) => {
+                span.record(WSTS_DKG_ID, request.dkg_id);
+                span.record(WSTS_SIGNER_ID, request.signer_id);
+                span.record(WSTS_SIGN_ID, request.sign_id);
+                span.record(WSTS_SIGN_ITER_ID, request.sign_iter_id);
+
+                let WstsMessageId::DkgVerification(key) = msg.id else {
+                    return Ok(());
+                };
+
+                let key = key.into();
+
+                let state_machine_id = self
+                    .ensure_dkg_verification_state_machine(chain_tip.block_hash, key)
+                    .await?;
```

Comment:
Ok now I remember the nuisance from the previous convo: if we allocate a dkg machine here (and in SignatureShareResponse), a rouge signer can send nonce response messages with message id some `DkgVerification(random_key)`: since we don't validate the key we end up creating new machines, potentially evicting legittimate ones from the lru cache.

We can address this after the buffering refactor, I just wanted to point this out. Or it could be enough to call `validate_dkg_verification_message` also here, maybe. Anyway, feel free to resolve this convo.

Reply:
I added validation to the key in [2c46ea5](https://github.com/stacks-network/sbtc/pull/1301/commits/2c46ea5ab57d2df6b0deb6983d51b58888111d9b) over on #1301 :)